src/main/java/io/cryostat/messaging/MessagingServer.java

```diff
@@ -153,13 +151,17 @@ public void start() throws SocketException, UnknownHostException {
                                                         sendClientActivityNotification(
                                                                 remoteAddress, "accepted");
 
-                                                        pingTask =
-                                                                this.keepalivePinger
-                                                                        .scheduleAtFixedRate(
-                                                                                wsc::ping,
-                                                                                0,
-                                                                                5,
-                                                                                TimeUnit.SECONDS);
+                                                        Long ping =
+                                                                pingTasks.put(
+                                                                        wsc,
+                                                                        vertx.setPeriodic(
+                                                                                TimeUnit.SECONDS
+                                                                                        .toMillis(
+                                                                                                5),
+                                                                                id -> wsc.ping()));
+                                                        if (ping != null) {
+                                                            vertx.cancelTimer(ping);
```

Comment:
Why is the periodic timer canceled immediately after creation?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#put-K-V-

> Returns: the previous value associated with key, or null if there was no mapping for key.

It will cancel any previous timer that existed for the same websocket client connection, but not the current timer. This probably never actually happens because it would require a second connection on the same ip/port pair to somehow be opened, without the first one having been closed and cleaned up first.